File: frontend/scripts/download-logos.js
/**
 * Node.js script to download NFL team logos
 * Run with: node scripts/download-logos.js
 */

const https = require('https');
const http = require('http');
const fs = require('fs');
const path = require('path');

const teams = [
  'ARI', 'ATL', 'BAL', 'BUF', 'CAR', 'CHI', 'CIN', 'CLE',
  'DAL', 'DEN', 'DET', 'GB', 'HOU', 'IND', 'JAX', 'KC',
  'LV', 'LAC', 'LAR', 'MIA', 'MIN', 'NE', 'NO', 'NYG',
  'NYJ', 'PHI', 'PIT', 'SF', 'SEA', 'TB', 'TEN', 'WAS'
];

// Create directory if it doesn't exist
const teamsDir = path.join(__dirname, '..', 'public', 'images', 'teams');
if (!fs.existsSync(teamsDir)) {
  fs.mkdirSync(teamsDir, { recursive: true });
}

// Function to download a file
function downloadFile(url, filepath) {
  return new Promise((resolve, reject) => {
    const protocol = url.startsWith('https') ? https : http;
    
    protocol.get(url, (response) => {
      if (response.statusCode === 200) {
        const fileStream = fs.createWriteStream(filepath);
        response.pipe(fileStream);
        fileStream.on('finish', () => {
          fileStream.close();
          resolve();
        });
      } else if (response.statusCode === 301 || response.statusCode === 302) {
        // Handle redirects
        downloadFile(response.headers.location, filepath)
          .then(resolve)
          .catch(reject);
      } else {
        reject(new Error(`Failed to download: ${response.statusCode}`));
      }
    }).on('error', reject);
  });
}

// Alternative URLs to try for each team
const getLogoUrls = (team) => {
  const teamLower = team.toLowerCase();
  return [
    `https://a.espncdn.com/i/teamlogos/nfl/500/${teamLower}.png`,
    `https://a.espncdn.com/i/teamlogos/nfl/500-dark/${teamLower}.png`,
    `https://static.www.nfl.com/image/private/t_headshot_desktop/league/${teamLower}`,
  ];
};

async function downloadLogos() {
  console.log('Downloading NFL team logos...\n');
  
  const results = { success: [], failed: [] };
  
  for (const team of teams) {
    const filepath = path.join(teamsDir, `${team}.png`);
    const urls = getLogoUrls(team);
    
    let downloaded = false;
    
    for (const url of urls) {
      try {
        await downloadFile(url, filepath);
        console.log(`✓ Downloaded ${team}`);
        results.success.push(team);
        downloaded = true;
        break;
      } catch (error) {
        // Try next URL
        continue;
      }
    }
    
    if (!downloaded) {
      console.log(`✗ Failed to download ${team}`);
      results.failed.push(team);
    }
  }
  
  console.log(`\nDone!`);
  console.log(`Successfully downloaded: ${results.success.length} logos`);
  console.log(`Failed: ${results.failed.length} logos`);
  
  if (results.failed.length > 0) {
    console.log(`\nFailed teams: ${results.failed.join(', ')}`);
    console.log('\nYou can manually download these logos from:');
    console.log('- https://www.espn.com/nfl/teams');
    console.log('- https://www.nfl.com/teams');
    console.log('- https://logos-world.net/nfl-logos/');
    console.log('\nSave them as {TEAM_ABBREV}.png in public/images/teams/');
  }
}

downloadLogos().catch(console.error);

File: frontend/scripts/download-logos.py
#!/usr/bin/env python3
"""
Python script to download NFL team logos
Run with: python3 scripts/download-logos.py
"""

import os
import urllib.request
from pathlib import Path

teams = [
    'ARI', 'ATL', 'BAL', 'BUF', 'CAR', 'CHI', 'CIN', 'CLE',
    'DAL', 'DEN', 'DET', 'GB', 'HOU', 'IND', 'JAX', 'KC',
    'LV', 'LAC', 'LAR', 'MIA', 'MIN', 'NE', 'NO', 'NYG',
    'NYJ', 'PHI', 'PIT', 'SF', 'SEA', 'TB', 'TEN', 'WAS'
]

# Create directory if it doesn't exist
script_dir = Path(__file__).parent
teams_dir = script_dir.parent / 'public' / 'images' / 'teams'
teams_dir.mkdir(parents=True, exist_ok=True)

def download_logo(team, url, filepath):
    """Download a logo from URL to filepath"""
    try:
        urllib.request.urlretrieve(url, filepath)
        return True
    except Exception as e:
        return False

def get_logo_urls(team):
    """Get possible URLs for a team logo"""
    team_lower = team.lower()
    return [
        f'https://a.espncdn.com/i/teamlogos/nfl/500/{team_lower}.png',
        f'https://a.espncdn.com/i/teamlogos/nfl/500-dark/{team_lower}.png',
    ]

print('Downloading NFL team logos...\n')

success_count = 0
failed_teams = []

for team in teams:
    filepath = teams_dir / f'{team}.png'
    urls = get_logo_urls(team)
    
    downloaded = False
    for url in urls:
        if download_logo(team, url, filepath):
            print(f'✓ Downloaded {team}')
            success_count += 1
            downloaded = True
            break
    
    if not downloaded:
        print(f'✗ Failed to download {team}')
        failed_teams.append(team)

print(f'\nDone!')
print(f'Successfully downloaded: {success_count} logos')
print(f'Failed: {len(failed_teams)} logos')

if failed_teams:
    print(f'\nFailed teams: {", ".join(failed_teams)}')
    print('\nYou can manually download these logos from:')
    print('- https://www.espn.com/nfl/teams')
    print('- https://www.nfl.com/teams')
    print('- https://logos-world.net/nfl-logos/')
    print(f'\nSave them as {{TEAM_ABBREV}}.png in {teams_dir}/')

